apps/web/src/app/[locale]/(profile)/[username]/_components/dashboard/_actions.ts

```diff
@@ -152,3 +152,81 @@ export async function getSolvedChallenges() {
     percentage,
   };
 }
+
+export interface BadgeInfo {
+  // eslint-disable-next-line @typescript-eslint/sort-type-constituents
+  slug: 'aot-2023-bronze' | 'aot-2023-silver' | 'aot-2023-gold' | 'aot-2023-platinum';
+  name: string;
+}
+
+export async function getBadges(userId: string): Promise<BadgeInfo[]> {
+  const badges: BadgeInfo[] = [];
+
+  const holidayTrack = await prisma.track.findFirstOrThrow({
+    where: {
+      slug: 'advent-of-typescript-2023',
+    },
+    include: {
+      trackChallenges: {
+        orderBy: {
+          orderId: 'asc',
+        },
+        include: {
+          challenge: {
+            include: {
+              submission: {
+                where: {
+                  userId,
+                },
+              },
+            },
+          },
+        },
+      },
+      enrolledUsers: {
+        where: {
+          id: userId,
+        },
+        select: {
+          id: true,
+        },
+      },
+    },
+  });
+
+  const numberOfInProgressHolidayChallenges = holidayTrack.trackChallenges.filter(
+    (trackChallenge) => {
+      return (
+        trackChallenge.challenge.submission.length &&
```

Comment:
do you have to check length here or can you just `?.every`

Reply:
sounds good, implemented. im not much of a `?.` fan in boolean logic (feels "gotcha-y" compared to a length check) but happy to conform.